qep-342-charts.md

```diff
@@ -0,0 +1,123 @@
+# QGIS Enhancement: Charts
+
+**Date** 2025-06-20
+**Author** Mathieu Pellerin (@nirvn)
+**Contact** mathieu@opengis.ch
+**Version** QGIS 4.0
+
+## Summary
+
+This QEP aims at addressing a long-standing functionality gap within QGIS: 
+out-of-the-box charting capabilities. This crucial means to data visualization 
+has long been requested and would be a great addition to the layout designer. 
+
+### Motivations
+
+By creating QGIS-powered charting classes instead of relying on 3rd-party 
+libraries (such as D3), we would unlock powerful styling by leveraging our 
+pre-existing symbology capabilities. 
+
+Furthermore, the rendered graphs would benefit from a tighter integration 
+with map canvas and other layout items through sharing the same symbols, 
+drawn using the same text rendering engine, etc. This also means the resulting 
+charts retain their vector state when exporting content in supported formats 
+(PDFs, SVGs, etc.) and will be CMYK-compatible when running on Qt6 builds.
+
+Finally, developing a set of chart classes in src/core will be beneficial 
+to the QGIS server project as well as other sister projects such as QField 
+where solutions based on python plugins or QtWebkit-driven renderings simply 
+do not work.
+
+### Proposed solution
+
+The proposed solution will add charting classes aimed at behaving similarly
+to symbol classes with a render function taking a render context object.
```

Comment:
```suggestion
to symbol classes with a render function taking a render context object, and an argument containing the data to render in the chart
```

Can I also suggest that we add a "QgsPlotRenderContext" object for future proofing the API (Eg akin to QgsCallout/QgsCalloutContext)? We've been bitten in the past by API stability and missing a place to specify extra object-specific stuff which doesn't fit into the generic QgsRenderContext class.

Reply:
@nyalldawson , do you want the QgsRenderContext parameter to be replaced by a new QgsPlotRenderContext there, which would currently hold a QgsRenderContext member?

Reply:
no, i'd just add it as a new argument separate from the QgsRenderContext

Reply:
OK, it'll be done.